features/step_wrapper_with_options.feature

```diff
@@ -0,0 +1,42 @@
+Feature: Step Wrapper with Options
+  In order to be able to write more complex step definition wrappers
+  As a developer
+  I want Cucumber to provide the "options" object to the wrapping function
+
+  Background:
+    Given a file named "features/a.feature" with:
+      """
+      Feature: Step with an option
+        Scenario: Steps
+          When I run a step with options
+      """
+    And a file named "features/step_definitions/cucumber_steps.js" with:
+      """
+      import {defineSupportCode} from 'cucumber'
+
+      defineSupportCode(({Then, When}) => {
+        When(/^I run a step with options$/, {retry: 2}, function () {
```

Comment:
The problem with non-standard options is if somewhere down the line cucumber adds a `retry` option and now that is used for multiple things. By having `wrapperOptions`, the content is specifically for the `setDefinitionFunctionWrapper` function and can never clash with future cucumber-js options